modules/dag-shared/src/main/scala/org/tessellation/dag/block/BlockValidator.scala

```diff
@@ -0,0 +1,153 @@
+package org.tessellation.dag.block
+
+import cats.Applicative
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.block.BlockValidator._
+import org.tessellation.dag.transaction.TransactionValidator
+import org.tessellation.dag.transaction.TransactionValidator.TransactionValidationError
+import org.tessellation.dag.transaction.filter.Consecutive
+import org.tessellation.dag.types.{BlockReference, DAGBlock}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{Transaction, TransactionReference}
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.Signed
+
+abstract class BlockValidator[F[_]](transactionValidator: TransactionValidator[F], consensusPeersCount: Int) {
+
+  implicit val F: Async[F]
+  implicit val kryoSerializer: KryoSerializer[F]
+  implicit val securityProvider: SecurityProvider[F]
```

Comment:
if you have an abstract class, you can put it into the class definition

Reply:
Right. After decision about abstract class vs trait I will refactor that.